### PR TITLE
[State CockroachDB] Adds Component Metadata Schema

### PR DESCRIPTION
**Pull request by robertojrojas**
# Description

[State CockroachDB] Adds Component Metadata Schema.

## Issue reference
This PR will close: #3054

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by artursouza — DISMISSED** [review]

File: state/cockroachdb/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      Can optionally have the schema name as prefix, such as public.dapr_metadata.
+    type: string
+    default: '"dapr_metadata"' 
+    example: '"dapr_metadata", "public.dapr_metadata"'
+  - name: cleanupIntervalInSeconds
+    description: |
+      Interval, in seconds, to clean up rows with an expired TTL.
+      Setting this to values <=0 disables the periodic cleanup.
+    type: number
+    default: '"3600"'
```
Comment (artursouza) [inline state/cockroachdb/metadata.yaml:54]: If just a number then no need for double quote

Comment (robertojrojas) [inline state/cockroachdb/metadata.yaml:54]: Validate that `"` are needed.

**Review by artursouza — APPROVED** [review]

File: state/cockroachdb/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    example: '30'
+  - name: tableName
+    description: |
+      Name of the table where the data is stored. Defaults to state. Can optionally have 
+      the schema name as prefix, such as public.state.
+    type: string
+    default: '"state"'  
+    example: '"state", "public.state"'
+  - name: metadataTableName
+    description: |
```
Comment (ItalyPaleAle) [inline state/cockroachdb/metadata.yaml:43]: Same comments as above

File: state/cockroachdb/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    type: number
+    default: '3600'
+    example: '1800, -1'
+  - name: connectionMaxIdleTime
+    description: |
+      Max idle time before unused connections are automatically closed in the connection pool.
+      By default, there’s no value and this is left to the database driver to choose.
+    type: duration
+    example: '"5m"'
+  - name: actorStateStore
```
Comment (ItalyPaleAle) [inline state/cockroachdb/metadata.yaml:62]: This is added automatically and should not be listed here

**Review by ItalyPaleAle — COMMENTED** [review]
@robertojrojas please fix the errors highlighted as soon as possible in a separate PR, or I’ll revert this.

CC: @artursouza

File: state/cockroachdb/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  - ttl
+  - actorStateStore
+authenticationProfiles:
+  - title: "Connection string"
+    description: "Authenticate using a Connection String"
+    metadata:
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: The connection string for the PostgreSQL database
```
[the remark this answers is not in the document]
Comment (robertojrojas) [reply state/cockroachdb/metadata.yaml:24]: This is related to this PR https://github.com/dapr/components-contrib/pull/2632

[the remark this answers is not in the document]
Comment (ItalyPaleAle) [reply state/cockroachdb/metadata.yaml:24]: Yes but this is the Cockraoch DB component, not Postgres. 2632 is an implementation detail

**Pull request merged**